boltdb/boltdb_manager_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package boltdb
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/timshannon/bolthold"
+)
+
+// TODO(temikus): Remove this helper in favour of T.Cleanup when we upgrade to Go 1.14+
+// See https://tip.golang.org/doc/go1.14#testing
+func CleanupStore(manager *Manager, records []string) error {
+	// datatype is just an example of the type stored so that the proper bucket and indexes are updated
+	datatype := StoredMetricRecord{}
+
+	query := bolthold.Where("Name").In(bolthold.Slice(records)...)
+
+	if err := manager.Store.DeleteMatching(datatype, query); err != nil {
+		return fmt.Errorf("unable to clean up records from BoltDB: %v", err)
+	}
+
+	return nil
+}
+
+func TestBoltdbManager(t *testing.T) {
+	ctx := context.Background()
+
+	// Create a temporary file for BoltDB
+	tempFile, err := ioutil.TempFile("", "boltdb")
+	if err != nil {
+		t.Fatalf("Unable to create a temporary file for BoltDB: %v", err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	manager := New(ctx, &Options{DBPath: tempFile.Name()})
+	defer manager.Close()
+
+	testMetrics := []string{"metric1", "metric2"}
+	defer CleanupStore(manager, []string{})
```

Comment:
I am not sure I understand why you need to do this given that the temporary database file will be removed anyway?

Reply:
Ah, that's a leftover. Removing it and the helper.